pkg/reconciler/pipeline/controller_test.go

```diff
@@ -862,3 +911,37 @@ func Test_copyVertexTemplate(t *testing.T) {
 		assert.Equal(t, "test", vtx.InitContainerTemplate.EnvFrom[0].SecretRef.Name)
 	})
 }
+
+func Test_checkChildrenResourceStatus(t *testing.T) {
+	t.Run("test check children resource status", func(t *testing.T) {
+		cl := fake.NewClientBuilder().Build()
+		ctx := context.TODO()
+		testIsbSvc := testNativeRedisIsbSvc.DeepCopy()
+		testIsbSvc.Status.MarkConfigured()
+		testIsbSvc.Status.MarkDeployed()
+		err := cl.Create(ctx, testIsbSvc)
+		assert.Nil(t, err)
+		r := &pipelineReconciler{
+			client:   cl,
+			scheme:   scheme.Scheme,
+			config:   reconciler.FakeGlobalConfig(t, fakeGlobalISBSvcConfig),
+			image:    testFlowImage,
+			logger:   zaptest.NewLogger(t).Sugar(),
+			recorder: record.NewFakeRecorder(64),
+		}
+		testObj := testPipelineWithSideinput.DeepCopy()
+		_, err = r.reconcile(ctx, testObj)
+		assert.NoError(t, err)
+		err = checkChildrenResourceStatus(ctx, cl, testObj)
+		assert.NoError(t, err)
+		for _, c := range testObj.Status.Conditions {
+			if c.Type == string(dfv1.PipelineConditionDaemonServiceHealthy) {
+				assert.Equal(t, string(corev1.ConditionTrue), string(c.Status))
+			} else if c.Type == string(dfv1.PipelineConditionSideInputServiceHealthy) {
+				assert.Equal(t, string(corev1.ConditionTrue), string(c.Status))
+			} else if c.Type == string(dfv1.PipelineConditionVerticesServiceHealthy) {
+				assert.Equal(t, string(corev1.ConditionFalse), string(c.Status))
```

Comment:
why PipelineConditionVerticesServiceHealthy corresponds to ConditionFalse instead of True?

Reply:
After reconciler ran, It's setting the vertex pod health as "False", Let me check the reason behind it.

Reply:
Here vertex reconciler is not running, so the status of vertex pod is not getting updated and getting set as false by default